frontend/src/components/chat/Chat.jsx

```diff
@@ -0,0 +1,211 @@
+      return;
+    }
+
+    connectSocket(user.token);
+    console.log("socket", socket);
+    return () => {
+      // Clean up socket on unmount
+      disconnectSocket();
+    };
+  }, [user.token]);
+
+  useEffect(() => {
+    const joinRoomAndGetChatHistory = async () => {
+      await joinDirectChatRoom(friend);
+      getChatHistory(friend);
+    };
+
+    if (friend) {
+      joinRoomAndGetChatHistory();
+    }
+  }, [friend]);
+
+  const sendMessage = async (message) => {
+    console.log("sending message", message);
+    if (socket) {
+      console.log("socket exists");
+      const res = await socket.emitWithAck("send-message", {
+        name: `${user.firstname} ${user.lastname}`,
+        message: message
+      });
+
+      if (res?.status === "Bad request") {
+        console.error("Bad request: ", res?.message);
+        return;
+      }
+      console.log(res);
+    }
+  };
+
+  useEffect(() => {
+    if (socket) {
+      socket.on("message", (message) => {
+        console.log("message received", message);
+        setMessages((prevMessages) => [...prevMessages, message]);
+      });
+
+      socket.on("message-history", (messageHistory) => {
+        console.log("message history received", messageHistory);
+        setMessages(messageHistory);
+      });
+
+      return () => {
+        socket.off("message");
+        socket.off("message-history");
+      };
+    } else {
+      console.error("No socket");
+    }
+  }, [user.token]);
+
+  return (
+    <div className="chat-container">
+      <div className="chat-container__chat-info">
+        <h1 className="chat-container__chat-info__title">Chat</h1>
+        {eventId && eventName && (
+          <div className="chat-container__chat-info__event-info">
+            <p>{eventName}</p>
+          </div>
+        )}
+        {friend && (
+          <div className="chat-container__chat-info__friend-info">
+            <p>
+              {friend.firstname} {friend.lastname}
+            </p>
+          </div>
+        )}
+      </div>
+      <div className="chat-container__messages-container">
+        {messages.map((message, index) => (
+          <div key={index} className="chat-container__message-container">
+            <MessageBubble
+              message={message}
+              isMine={message.senderUserId === user.id}
+              isGroupChat={Boolean(eventId)}
+            />
+          </div>
+        ))}
+      </div>
+      <div className="chat-container__bottom-area">
+        {isDisabled && (
+          <p className="chat-container__bottom-area__disabled-message">
+            {disabledMessage}
+          </p>
+        )}
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            const message = e.target.message.value;
+            sendMessage(message);
+            e.target.message.value = "";
+          }}
+        >
+          <input
+            data-testid="message-box"
+            type="text"
+            name="message"
```

Comment:
```suggestion
            name="message"
            placeholder="Send a message"
```

A placeholder message could be added here to match the Figma prototype.